File: app/src/main/java/com/fthdgn/books/repository/AuthorConverter.java
package com.fthdgn.books.repository;

import android.arch.persistence.room.TypeConverter;

import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("WeakerAccess")
public class AuthorConverter {

    @TypeConverter
    public static List<String> authorsFromString(String value) {
        String[] values = value.split("\\|");
        List<String> authors = new ArrayList<>(values.length);
        for (String v : values) {
            if (v.length() > 0) {
                authors.add(v);
            }
        }
        return authors;
    }

    @TypeConverter
    public static String authorsToString(List<String> values) {
        if (values == null)
            return "";
        StringBuilder stringBuilder = new StringBuilder();
        for (String value : values) {
            stringBuilder.append("|").append(value).append("|");
        }
        return stringBuilder.toString();
    }

}

File: app/src/main/java/com/fthdgn/books/repository/AuthorConverter.kt
package com.fthdgn.books.repository

import android.arch.persistence.room.TypeConverter
import java.util.*

class AuthorConverter {

    @TypeConverter
    fun authorsFromString(value: String): List<String> {
        val values = value.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val authors = ArrayList<String>(values.size)
        values.filterTo(authors) { it.isNotEmpty() }
        return authors
    }

    @TypeConverter
    fun authorsToString(values: List<String>?): String {
        if (values == null)
            return ""
        val stringBuilder = StringBuilder()
        for (value in values) {
            stringBuilder.append("|").append(value).append("|")
        }
        return stringBuilder.toString()
    }

}
